spring-integration-core/src/main/java/org/springframework/integration/support/json/Jackson3JsonObjectMapper.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2025-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.json;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.Reader;
+import java.io.Writer;
+import java.lang.reflect.Type;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import tools.jackson.core.JacksonException;
+import tools.jackson.core.JsonParser;
+import tools.jackson.databind.JacksonModule;
+import tools.jackson.databind.JavaType;
+import tools.jackson.databind.JsonNode;
+import tools.jackson.databind.ObjectMapper;
+import tools.jackson.databind.json.JsonMapper;
+
+import org.springframework.integration.mapping.support.JsonHeaders;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Jackson 3 JSON-processor (@link https://github.com/FasterXML)
+ * {@linkplain JsonObjectMapper} implementation.
+ * Delegates {@link #toJson} and {@link #fromJson}
+ * to the {@linkplain ObjectMapper}
+ * <p>
+ * It customizes Jackson's default properties with the following ones:
+ * <ul>
+ * <li>The well-known modules are registered through the classpath scan</li>
+ * </ul>
+ *
+ * See {@code tools.jackson.databind.json.JsonMapper.builder} for more information.
+ *
+ * @author Jooyoung Pyoung
+ *
+ * @since 7.0
+ *
+ */
+public class Jackson3JsonObjectMapper extends AbstractJacksonJsonObjectMapper<JsonNode, JsonParser, JavaType> {
+
+	private static final boolean JODA_MODULE_PRESENT =
+			ClassUtils.isPresent("tools.jackson.datatype.joda.JodaModule", null);
+
+	private static final boolean KOTLIN_MODULE_PRESENT =
+			ClassUtils.isPresent("kotlin.Unit", null) &&
+					ClassUtils.isPresent("tools.jackson.module.kotlin.KotlinModule", null);
+
+	private final ObjectMapper objectMapper;
+
+	public Jackson3JsonObjectMapper() {
+		List<JacksonModule> jacksonModules = collectWellKnownModulesIfAvailable();
+		this.objectMapper = JsonMapper.builder()
+				.addModules(jacksonModules)
+				.build();
+	}
+
+	public Jackson3JsonObjectMapper(ObjectMapper objectMapper) {
+		Assert.notNull(objectMapper, "objectMapper must not be null");
+		this.objectMapper = objectMapper;
+	}
+
+	public ObjectMapper getObjectMapper() {
+		return this.objectMapper;
+	}
+
+	@Override
+	public String toJson(Object value) throws JacksonException {
```

Comment:
I think to support this abstraction contract we have to re-throw an `IOException` instead.